source/agora/node/BlockStorage.d

```diff
@@ -932,8 +924,9 @@ public class MemBlockStorage : IBlockStorage
 
     public void saveBlock (const ref Block block) @safe
     {
-        if (this.blocks.length != block.header.height)
-            throw new Exception("BlockStorage: Expected blocks in serial order");
+        enforce(this.blocks.length == block.header.height,
+                "BlockStorage: Expected blocks in serial order (known: {}, got: {})",
+                this.blocks.length, block.header.height,);
```

Comment:
I took the liberty to add some information to those, since I went with `git grep -n 'throw new'`